vsutil/__init__.py

```diff
@@ -251,6 +251,65 @@ def is_image(filename: str, /) -> bool:
     return types_map.get(path.splitext(filename)[-1], '').startswith('image/')
 
 
+def scale(value: float, 
+          input_depth: int, 
+          output_depth: int, 
+          range_in: Union[int, Range] = 0, 
+          range: Optional[Union[int, Range]] = None, 
+          scale_offsets: bool = False, 
+          chroma: bool = False) -> float:
+    """
+    Scales a given value between bit depths, sample types, and/or ranges.
+    :value:         Numeric value to be scaled
+    :input_depth:   Bit depth of the "value" parameter. Use 32 for float samples
+    :output_depth:  Bit depth to scale the input value to
+    :range_in:      Pixel range of the input value. No clamping is performed
+    :range:         Pixel range of the output value. No clamping is performed
+    :scale_offsets: Whether or not to apply YUV offsets to float chroma and/or TV range integer values
+                    e.g. when scaling a TV range value of 16 to float, setting this to True will return "0.0" rather than "0.073059.."
+    :chroma:        Whether to treat values as chroma instead of luma
+    """
+
+    range_in = _resolve_enum(Range, range_in, 'range_in')
+    range = _resolve_enum(Range, range, 'range')
+    range = fallback(range, range_in)
+
+    if input_depth == 32:
+        input_peak = 1
+        range_in = 1
+    elif range_in:
+        input_peak = (1 << input_depth) - 1
+    else:
+        input_peak = (224 if chroma else 219) << (input_depth - 8)
```

Comment:
If my memory (and also Wikipedia) serves me right, the peak values for limited range are 235 for luma and 240 for chroma. Am I missing something here?

Reply:
Those are the peaks with no YUV offsets applied, so 235 - 16 and 240 - 16